### PR TITLE
Only accept Data Response when Network Data, Active, and Pending Datasets are known.

### PR DESCRIPTION
**Pull request by jwhui**

### CODE REVIEW

File: src/core/thread/mle.cpp [hunk 1]
```diff
@@ -1725,51 +1720,82 @@ ThreadError Mle::HandleDataResponse(const Message &aMessage, const Ip6::MessageI
...
-        }
-        else
-        {
-            dataRequest = true;
+            ExitNow(dataRequest = true);
         }
     }
+    else
+    {
+        activeTimestamp.SetLength(0);
```
Comment (luwannest) [inline src/core/thread/mle.cpp]: Does it mean that Data Response may always try to include timestamp? 
Not including timestamp indicates that does not have active/pending dataset and the node needs to clear the copy of network dataset.

Comment (jwhui) [inline src/core/thread/mle.cpp]: If the Data Response has no Active Timestamp TLV, then we remember that by setting the `activeTimestamp` length to zero.  Further below where we actually perform the updates to the Network Data, Active Dataset, and Pending Datasets, if the length of `activeTimestamp` is zero, then we clear the copy of the network dataset.

File: src/core/thread/mle.cpp [hunk 2]
```diff
@@ -1725,51 +1720,82 @@ ThreadError Mle::HandleDataResponse(const Message &aMessage, const Ip6::MessageI
...
+    mRetrieveNewNetworkData = false;
+
 exit:
     (void)aMessageInfo;
 
-    if (diff > 0 && dataRequest)
+    if (dataRequest)
     {
-        uint8_t tlvs[] = {Tlv::kNetworkData};
+        static const uint8_t tlvs[] = {Tlv::kNetworkData};
```
Comment (luwannest) [inline src/core/thread/mle.cpp]: One more question here, do we need to always include kLeaderData in request TLV?
Harness checks this, but I only found kNetworkData in spec.

Comment (jwhui) [inline src/core/thread/mle.cpp]: I don't see any place in the spec that requires including the Leader Data TLV in a MLE Data Request.

**Comment by luwannest** [comment]
LGTM

**Pull request merged**